app/src/ui/lib/text-box.tsx

```diff
@@ -116,6 +115,24 @@ export class TextBox extends React.Component<ITextBoxProps, ITextBoxState> {
     }
   }
 
+  public selectAll() {
+    if (this.inputElement == null) {
```

Comment:
`===`?

Reply:
`==` checks `=== null || === undefined`.

Reply:
We want to check for `null` and `undefined` here. If we did strict equality here, we could try to call `.select()` on `undefined`, and that can be real bad. Using `==` allows us to check for both without the much longer `x === null || x === undefined`.

Reply:
I don't see how `inputElement` could ever be undefined though? It's typed as `HTMLInputElement | null`. In what scenario are we expecting it to be `undefined`? If there's such a scenario we should type it accordingly

Reply:
Actually, it will never be undefined because I initialized it with a value of `null`. So yea, we should use the stronger check here. 

For fun, here's a program where I have told TS that a variable cannot be undefined, yet somehow, the variable is undefined. Maybe it's an opportunity for the TS guys to prevent a variable that can't be `undefined` to be declared without an assignment.

```ts
let cantBeUndefined: number | null; //maybe TS should say no here???

cantBeUndefined //undefined
```

Reply:
@iAmWillShepherd Defining the variable without assigning it is okay in that scenario but attempting to use it before assigning it should yield an error

![screen shot 2018-02-27 at 18 21 13](https://user-images.githubusercontent.com/634063/36743765-588fa540-1beb-11e8-96b6-c19651f163a8.png)

For the property TypeScript recently added tougher checks for unassigned properties but we'll have to wait on your upgrade PR (#3902) to make use of them.

Reply:
That's just the thing tho, I didn't produce an error for me.

![2018-02-28_00-11-02 293](https://user-images.githubusercontent.com/1715082/36772705-ed3dfd1a-1c1b-11e8-87ba-fd152da4c68f.png)

I'm using 2.6.2 in the REPL I produced this on. Maybe VS Cdoe isn't surfacing the error somehow???